zerver/views/message_fetch.py

```diff
@@ -957,6 +957,7 @@ def get_messages_backend(
                 MAX_MESSAGES_PER_FETCH,
             )
         )
+    realm = get_valid_realm_from_request(request)
```

Comment:
I guess moving this here doesn't cost a database query, since it does the `request.user` thing if possible.  OK.